cartography/intel/gcp/cloudcdn.py

```diff
@@ -0,0 +1,530 @@
+import json
+import logging
+from typing import Dict
+from typing import List
+
+import time
+import neo4j
+from googleapiclient.discovery import HttpError
+from googleapiclient.discovery import Resource
+
+from cartography.util import run_cleanup_job
+from . import label
+from cartography.util import timeit
+
+logger = logging.getLogger(__name__)
+
+@timeit
+def get_backend_buckets(compute: Resource, project_id: str) -> List[Dict]:
+    backend_buckets = []
+    try:
+        req = compute.backendBuckets().list(project=project_id)
+        while req is not None:
+            res = req.execute()
+            if res.get('items'):
+                for bucket in res['items']:
+                    bucket['region'] = 'global'
+                    bucket['id'] = f"projects/{project_id}/global/backendBuckets/{bucket['bucketName']}"
+                    backend_buckets.append(bucket)
+            req = compute.backendBuckets().list_next(previous_request=req, previous_response=res)
+
+        return backend_buckets
+    except HttpError as e:
+        err = json.loads(e.content.decode('utf-8'))['error']
+        if err.get('status', '') == 'PERMISSION_DENIED' or err.get('message', '') == 'Forbidden':
+            logger.warning(
+                (
+                    "Could not retrieve backend buckets on project %s due to permissions issues. Code: %s, Message: %s"
+                ), project_id, err['code'], err['message'],
+            )
+            return []
+        else:
+            raise
+
+@timeit
+def load_backend_buckets(session: neo4j.Session, buckets: List[Dict], project_id: str, update_tag: int) -> None:
+    session.write_transaction(load_backend_buckets_tx, buckets, project_id, update_tag)
+
+@timeit
+def load_backend_buckets_tx(
+    tx: neo4j.Transaction, buckets: List[Dict],
+    project_id: str, gcp_update_tag: int,
+) -> None:
+    query = """
+    UNWIND {Buckets} as b
+    MERGE (bucket:GCPBackendBucket{id:b.id})
+    ON CREATE SET
+        bucket.firstseen = timestamp()
+    SET
+        bucket.lastupdated = {gcp_update_tag},
```

Comment:
We should have a unique id (like arn in aws) for all objects that we are inserting into the neo4j db. We have followed something like this in our Audit project.